stubs/Flask-SocketIO/flask_socketio/__init__.pyi

```diff
@@ -0,0 +1,97 @@
+from collections.abc import Callable
+from logging import Logger
+from threading import Thread
+from typing import Any
+from typing_extensions import NotRequired, ParamSpec, TypeAlias, TypedDict, Unpack
+
+from flask import Flask
+from flask.testing import FlaskClient
+from socketio.base_manager import BaseManager  # type: ignore
+
+from .namespace import Namespace
+from .test_client import SocketIOTestClient
+
+_P = ParamSpec("_P")
+_ExceptionHandler: TypeAlias = Callable[[Exception], Any]
+_Handler: TypeAlias = Callable[[Any], Any]
+
+_HandlerDecorator: TypeAlias = Callable[[_Handler], _Handler]
+
+gevent_socketio_found: bool
+
+class _SocketIOConfig(TypedDict, total=False):
+    manage_session: NotRequired[bool]
+    message_queue: NotRequired[str]
+    channel: NotRequired[str]
+    path: NotRequired[str]
+    resource: NotRequired[str]
+
+    # SocketIO.Server options
+    client_manager: NotRequired[BaseManager]
+    logger: NotRequired[bool | Logger]
+    json: NotRequired[Any]
+    async_handlers: NotRequired[bool]
+    always_connect: NotRequired[bool]
+
+class _SocketIOCallArgs(TypedDict, total=False):
+    namespace: NotRequired[str]
+    to: NotRequired[str]
+    room: NotRequired[str]
+
+class _SocketIOEmitArgs(_SocketIOCallArgs, total=False):
+    include_self: NotRequired[bool]
+    skip_sid: NotRequired[str | list[str]]
+    callback: NotRequired[Callable[..., Any]]
+
+class _SocketIORunArgs(TypedDict, total=False):
+    debug: NotRequired[bool]
+    use_reloader: NotRequired[bool]
+    reloader_options: dict[str, Any]
+    log_output: NotRequired[bool]
+    allow_unsafe_werkzeug: NotRequired[bool]
+
+class SocketIO:
+    def __init__(self, app: Flask | None = None, **kwargs: Unpack[_SocketIOConfig]) -> None: ...  # type: ignore
```

Comment:
Unfortunately, PEP 692 cannot yet be used in typeshed. See #9710 for details.

Reply:
@AlexWaygood so then how should I type hint the kwargs? Or should I just remove all type hints for kwargs?

Reply:
Unfortunately, the best thing to do for now is probably to write out each signature in full, even though this is pretty tedious and repetitive. So for this function, you'd do it like this:

```suggestion
    def __init__(
        self, app: Flask | None = None,
        *,
        manage_session: bool = ...,
        message_queue: str = ...,
        channel: str = ...,
        path: str = ...,
        resource: str = ...,
        # SocketIO.Server options
        client_manager: BaseManager = ...,
        logger: bool | Logger = ...,
        json: Any = ...,
        async_handlers: bool = ...,
        always_connect: bool = ...,
    ) -> None: ...
```

And you would have to do the same for all functions where `**kwargs` are currently annotated with `Unpack[_SocketIOConfig]`.

Reply:
@AlexWaygood As you suggested, I have explicitly added type hints for all `**kwargs` without using `Unpack`.